### DISCUSSION TITLE
Automatic "Created at" & "Last edited" fields?

### DISCUSSION THREAD

**Discussion by lokimckay** [tinacms/tinacms#3551] [Help] [answered]
I want to add fields for the `created` and `last edited` datetimes that automatically update when a post is created/updated

I imagine I would need to hook into the submit action of Tina's sidebar somehow but after searching through the [docs](https://tina.io/docs/) many times I couldn't figure out how to do this

Can anyone please point me in the right direction?

**Comment by lokimckay** [comment, marked as answer]
You can at least get a created date to populate when a document is created using the ["defaultItem"](https://tina.io/docs/schema/#defining-defaults) key like this:

```js
export default {
  name: "post",
  label: "Posts",
  path: "content/blog",
  format: "mdx",
  defaultItem: () => ({
      created: new Date().toISOString(),
      lastEdited: new Date().toISOString(),
// ...
```

It doesn't seem like it is possible to add custom logic when a document is saved at the moment

![image](https://user-images.githubusercontent.com/14341953/215917422-c1104735-dae4-4a24-8516-51c395f0fd70.png)

If we could update the fields of documents when they are saved, this might facilitate updating the "last edited" date

Any plans/thoughts about this feature? @jamespohalloran

**Reply by Perni1984** [reply]
Although I have not tried it yet myself, IMHO this can be solved via the parse function of the field:
https://tina.io/docs/extending-tina/format-and-parse/

**Reply by Fox32** [reply]
Just using `parse` doesn't work well for a last modified field. But I created a custom field for it:

```tsx
import { useEffect } from 'react';
import { TinaField } from 'tinacms';

export function LastModifiedField({
  form,
  field,
}: {
  field: TinaField & { namespace: string[] };
  // TODO: Is this type available somewhere? External library?
  form: {
    subscribe: (
      callback: (opts: { dirty: boolean }) => void,
      opts: { dirty: boolean }
    ) => () => void;
    change: (name: string, value: string) => void;
  };
}) {
  useEffect(() => {
    return form.subscribe(
      ({ dirty }) => {
        if (dirty) {
          form.change(field.name, new Date().toISOString());
        }
      },
      { dirty: true }
    );
  }, [field.name, form]);

  return null;
}
```
